testing/e2e/agent_install_test.go

```diff
@@ -169,122 +150,107 @@ func (suite *AgentInstallSuite) extractZip(r io.Reader) {
 	suite.Require().NoError(err)
 	zipReader, err := zip.NewReader(bytes.NewReader(b.Bytes()), n)
 	suite.Require().NoError(err)
-	fleetPath := ""
 	for _, file := range zipReader.File {
-		if file.FileInfo().IsDir() {
-			err := os.Mkdir(filepath.Join(suite.downloadPath, file.Name), 0755)
-			suite.Require().NoError(err)
-		} else {
-			dst, err := os.Create(filepath.Join(suite.downloadPath, file.Name))
-			suite.Require().NoError(err)
-			err = dst.Chmod(0755)
+		path := filepath.Join(suite.downloadPath, file.Name)
+		mode := file.FileInfo().Mode()
+		switch {
+		case mode.IsDir():
+			err := os.MkdirAll(path, 0755)
```

Comment:
I understand that this will run on windows (it's a .zip file) but maybe we could try to use the actual permissions we have in the zip archive ?

Reply:
i don't see this as critical, chmod has very little effect on windows unless it's `0444`